src/main/java/frc/team2412/robot/commands/indexer/IndexBitmapCommand.java

```diff
@@ -0,0 +1,168 @@
+
+    public final int validBits;
+    public final int expectedBits;
+    private final IndexDirection intakeOffFrontIndexDir;
+    private final IndexDirection intakeOffBackIndexDir;
+    private final IndexDirection intakeOnFrontIndexDir;
+    private final IndexDirection intakeOnBackIndexDir;
+
+    // Get index motor directions for a given Intake state
+    public IndexDirection getFrontIndexDirection(final boolean intakeOn) {
+      return intakeOn ? intakeOnFrontIndexDir : intakeOffFrontIndexDir;
+    }
+
+    public IndexDirection getBackIndexDirection(final boolean intakeOn) {
+      return intakeOn ? intakeOnBackIndexDir : intakeOffBackIndexDir;
+    }
+
+    private IndexCommandEntry(final int validBits, final int expectedBits, final IndexDirection intakeOffFrontIndexDir, final IndexDirection intakeOffBackIndexDir,
+    final IndexDirection intakeOnFrontIndexDir, final IndexDirection intakeOnBackIndexDir) {
+      assert ((validBits & expectedBits) == expectedBits);
+      this.validBits = validBits;
+      this.expectedBits = expectedBits;
+      this.intakeOffFrontIndexDir = intakeOffFrontIndexDir;
+      this.intakeOffBackIndexDir = intakeOffBackIndexDir;
+      this.intakeOnFrontIndexDir = intakeOnFrontIndexDir;
+      this.intakeOnBackIndexDir = intakeOnBackIndexDir;
+    }
+  }
+
+  public IndexBitmapCommand(final IndexerMotorSubsystem indexerMotorSubsystem,
+      final IndexerSensorSubsystem indexerSensorSubsystem) {
+    m_indexerMotorSubsystem = indexerMotorSubsystem;
+    m_indexerSensorSubsystem = indexerSensorSubsystem;
+    addRequirements(indexerMotorSubsystem);
+  }
+
+  @Override
+  public void execute() {
+    IntakeDirection intakeDirection = RobotState.getintakeDirection();
+    final boolean intakeOn = (intakeDirection != IntakeDirection.NONE);
+    if (intakeDirection == IntakeDirection.NONE) {
+      if (s_lastIntakeDirection == IntakeDirection.BACK) {
+        intakeDirection = IntakeDirection.BACK;
+      } else {
+        intakeDirection = IntakeDirection.FRONT;
+      }
+    } else {
+      s_lastIntakeDirection = intakeDirection;
+    }
+
+    int sensorBitmap = 0;
+    switch (intakeDirection) {
+    case BOTH:
+    case FRONT:
+      sensorBitmap = (m_indexerSensorSubsystem.getSensorBitmapFrontLSB() & VALID_SENSOR_BITS);
+      break;
+    case BACK:
+      sensorBitmap = (m_indexerSensorSubsystem.getSensorBitmapBackLSB() & VALID_SENSOR_BITS);
+      break;
+    case NONE:
+      assert (false);
+      break;
+    }
+
+    IndexCommandEntry indexCommand = null;
+    for (final IndexCommandEntry entry : IndexCommandEntry.values()) {
+      if (entry.expectedBits == (sensorBitmap & entry.validBits)) {
+        indexCommand = entry;
+      }
+    }
+
+    if (indexCommand != null) {
+      final IndexDirection frontIndexDirection = indexCommand.getFrontIndexDirection(intakeOn);
+      final IndexDirection backIndexDirection = indexCommand.getBackIndexDirection(intakeOn);
+
```

Comment:
This should also detect & handle the case where the lookup table says 'Intake needs to be turned off because inputting a ball will cause us to jam or eject a ball out the other side', but I'm not sure the best way to do that.  That's left as an exercise for the reader...